### ISSUE TITLE
[Bug]: Multiple errors on initOneSignal SDK

### ISSUE THREAD

**Issue by IvanGarza07** [OneSignal/OneSignal-Android-SDK#1869]
### What happened?

The sdk present these errors since version 5.0.0:
- java.lang.Exception - Unrecognized operation: login-user-from-subscription
- org.json.JSONException - Value at lon of type java.lang.String cannot be converted to long
- java.lang.NullPointerException

these happen when execute the OneSignal.initWithContext(context, BuildConfig.ONE_SIGNAL_KEY)

### Steps to reproduce?

```Markdown
i dont have idea to reproduce the error
```

### What did you expect to happen?

i expected the SDK initiate normally

### OneSignal Android SDK version

5.0.0 and above

### Android version

13, 12, 11, 10, 9, 8

### Specific Android models

```Markdown
- Samsung Galaxy A10e
- Moto E30
- Samsung Galaxy S8+
- Samsung Galaxy A5
- Model:Realme C21Y
- Poco M4 Pro
- Redmi Note 10 Pro
- Model:Alcatel 3
- Galaxy A10s
- Galaxy S9
- Xiaomi 11T
- Galaxy A32
- Model:TCL 30XE
- Galaxy Note8
- Moto E40
- Motorola One Fusion
- Huawei MediaPad T3 10
```

### Relevant log output

```Shell
java.lang.reflect.Constructor.newInstance0 (Constructor.java)
java.lang.reflect.Constructor.newInstance (Constructor.java:343)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:90)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:83)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:424)
com.onesignal.OneSignal.initWithContext (OneSignal.kt:126)

com.onesignal.core.internal.operations.impl.OperationModelStore.create (OperationModelStore.kt:58)
com.onesignal.core.internal.operations.impl.OperationModelStore.create (OperationModelStore.kt:27)
com.onesignal.common.modeling.ModelStore.load (ModelStore.kt:129)
com.onesignal.core.internal.operations.impl.OperationModelStore.<init> (OperationModelStore.kt:30)
java.lang.reflect.Constructor.newInstance0 (Constructor.java)
java.lang.reflect.Constructor.newInstance (Constructor.java:343)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:90)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:83)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:424)
com.onesignal.OneSignal.initWithContext (OneSignal.kt:126)

org.json.JSON.typeMismatch (JSON.java:101)
org.json.JSONObject.getLong (JSONObject.java:526)
com.onesignal.common.modeling.Model.initializeFromJson (Model.kt:103)
com.onesignal.user.internal.properties.PropertiesModel.createModelForProperty (PropertiesModel.kt:94)
com.onesignal.common.modeling.Model.initializeFromJson (Model.kt:86)
com.onesignal.common.modeling.SimpleModelStore.create (SimpleModelStore.kt:32)
com.onesignal.common.modeling.ModelStore.load (ModelStore.kt:129)
com.onesignal.common.modeling.SimpleModelStore.<init> (SimpleModelStore.kt:25)
com.onesignal.user.internal.properties.PropertiesModelStore.<init> (PropertiesModelStore.kt:8)
java.lang.reflect.Constructor.newInstance0 (Constructor.java)
java.lang.reflect.Constructor.newInstance (Constructor.java:343)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:90)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:83)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:83)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:419)
com.onesignal.OneSignal.initWithContext (OneSignal.kt:126)

com.onesignal.core.internal.application.impl.ApplicationService.getAppContext (ApplicationService.kt:39)
com.onesignal.core.internal.preferences.impl.PreferencesService.getSharedPrefsByName (PreferencesService.kt:157)
com.onesignal.core.internal.preferences.impl.PreferencesService.get (PreferencesService.kt:62)
com.onesignal.core.internal.preferences.impl.PreferencesService.getString (PreferencesService.kt:36)
com.onesignal.common.modeling.ModelStore.load (ModelStore.kt:126)
com.onesignal.common.modeling.SimpleModelStore.<init> (SimpleModelStore.kt:25)
com.onesignal.core.internal.config.ConfigModelStore.<init> (ConfigModelStore.kt:8)
java.lang.reflect.Constructor.newInstance0 (Constructor.java)
java.lang.reflect.Constructor.newInstance (Constructor.java:343)
com.onesignal.common.services.ServiceRegistrationReflection.resolve (ServiceRegistration.kt:90)
com.onesignal.common.services.ServiceProvider.getServiceOrNull (ServiceProvider.kt:79)
com.onesignal.common.services.ServiceProvider.getService (ServiceProvider.kt:67)
com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:417)
com.onesignal.OneSignal.initWithContext (OneSignal.kt:126)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by evgentset** [comment]
Observed similar issue on Redmi 5 Plus, Android 8.1.0,
My code is simple, just use init in onCreate method in Application class, like this;
 OneSignal.initWithContext(this, appId)

Logs:
```
10-17 13:30:55.914 29034 29034 E AndroidRuntime: Process: com.*.*, PID: 29034
10-17 13:30:55.914 29034 29034 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.*.*.MyApplication: java.lang.reflect.InvocationTargetException
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5829)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.ActivityThread.-wrap1(Unknown Source:0)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1685)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:176)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6651)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance0(Native Method)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:334)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:90)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:83)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:83)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:419)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.OneSignal.initWithContext(OneSignal.kt:126)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.*.*.MyApplication.onCreate(MyApplication.kt:25)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1120)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5826)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	... 8 more
10-17 13:30:55.914 29034 29034 E AndroidRuntime: Caused by: org.json.JSONException: Value test3 at key of type java.lang.String cannot be converted to boolean
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at org.json.JSON.typeMismatch(JSON.java:100)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at org.json.JSONObject.getBoolean(JSONObject.java:416)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.modeling.Model.initializeFromJson(Model.kt:107)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.user.internal.properties.PropertiesModel.createModelForProperty(PropertiesModel.kt:94)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.modeling.Model.initializeFromJson(Model.kt:87)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.modeling.SimpleModelStore.create(SimpleModelStore.kt:32)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:129)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.common.modeling.SimpleModelStore.<init>(SimpleModelStore.kt:25)
10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	at com.onesignal.user.internal.properties.PropertiesModelStore.<init>(PropertiesModelStore.kt:8)10-17 13:30:55.914 29034 29034 E AndroidRuntime: 	... 24 more
```

**Comment by emawby** [comment]
@IvanGarza07 Thank you for reaching out many of these issues have been resolved by releases since 5.0.0 although there may be still be existing issues. If you are able to reproduce any of these issues on the latest release any reproduction steps would be helpful

**Comment by IvanGarza07** [comment]
> @IvanGarza07 Thank you for reaching out many of these issues have been resolved by releases since 5.0.0 although there may be still be existing issues. If you are able to reproduce any of these issues on the latest release any reproduction steps would be helpful

i updated to latest version (5.0.3) but the errors persist, i try to reproduce the errors but can't reproduce the errors i dont have idea how to do that. I had to downgrade oneSignal to 4.8.3 version

**Comment by evgentset** [comment]
@emawby one more note, we're using Proguard/R8 in the app, maybe this related to the issue somehow. And the issue is reproducible not always, around 70% probably, sometimes it did not crashed the app.

**Comment by brismithers** [comment]
I believe there are a couple distinct issues and want to make sure nothing is missed:

**JSONException 'Value test3 at key..' and 'Value at lon...'**
I believe we have a fix for this in #1884. 


**java.lang.Exception - Unrecognized operation: login-user-from-subscription***
This looks to have been fixed by #1845 and is in v5.0.2


**java.lang.NullPointerException**
I don't see a stack trace for this one.  If you are still seeing this can you please provide more details and context around this one?

**Comment by brismithers** [comment]
Following up on this, #1884 was released in [v5.0.4](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.4).  Please create another issue if you see the NullPointerException again, thanks!
